### PR TITLE
TiUP: CDC support Data dir .

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

CDC support data dir for cluster version >= `v4.0.13`

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by overvenus** [comment]
Could you update `config-templates/complex-cdc.yml` too?

rest LGTM

File: tiup/tiup-cluster-topology-reference.md [hunk 1]
```diff
@@ -380,6 +380,7 @@ drainer_servers:
 - `ssh_port`：指定连接目标机器进行操作的时候使用的 SSH 端口，若不指定，则使用 `global` 区块中的 `ssh_port`
 - `port`：TiCDC 服务的监听端口，默认 8300
 - `deploy_dir`：指定部署目录，若不指定，或指定为相对目录，则按照 `global` 中配置的 `deploy_dir` 生成
+- `data_dir`：指定数据目录，若不指定，或指定为相对目录，则按照 `global` 中配置的 `data_dir` 生成。集群版本 `v4.0.13` 为最小支持版本，当集群版本号小于 `v4.0.13` 时将不会生效。
```
Comment (amyangfei) [inline tiup/tiup-cluster-topology-reference.md:383]:
```suggestion
- `data_dir`：指定数据目录，若不指定，或指定为相对目录，则按照 `global` 中配置的 `data_dir` 生成。集群版本 `v4.0.14`/`v5.0.2` 为最小支持版本，当集群版本号小于 `v4.0.14`/`v5.0.2` 时将不会生效。
```

**Comment by TomShawn** [comment]
@amyangfei Please help confirm whether this PR applies to v5.1. Thanks~

File: tiup/tiup-cluster-topology-reference.md [hunk 2]
```diff
@@ -380,6 +380,7 @@ drainer_servers:
 - `ssh_port`：指定连接目标机器进行操作的时候使用的 SSH 端口，若不指定，则使用 `global` 区块中的 `ssh_port`
 - `port`：TiCDC 服务的监听端口，默认 8300
 - `deploy_dir`：指定部署目录，若不指定，或指定为相对目录，则按照 `global` 中配置的 `deploy_dir` 生成
+- `data_dir`：指定数据目录，若不指定，或指定为相对目录，则按照 `global` 中配置的 `data_dir` 生成。集群版本 `v4.0.14`/`v5.0.2` 为最小支持版本，当集群版本号小于 `v4.0.14`/`v5.0.2` 时将不会生效。
```
Comment (TomShawn) [inline tiup/tiup-cluster-topology-reference.md:383]:
```suggestion
- `data_dir`：指定数据目录。若不指定该字段或指定为相对目录，数据目录则按照 `global` 中配置的 `data_dir` 生成。该字段自集群版本 v4.0.14/v5.0.2 起生效。在早于该版本的集群中，该字段不生效。
```

**Review by amyangfei — APPROVED** [review]

File: tiup/tiup-cluster-topology-reference.md [hunk 3]
```diff
@@ -380,6 +380,7 @@ drainer_servers:
 - `ssh_port`：指定连接目标机器进行操作的时候使用的 SSH 端口，若不指定，则使用 `global` 区块中的 `ssh_port`
 - `port`：TiCDC 服务的监听端口，默认 8300
 - `deploy_dir`：指定部署目录，若不指定，或指定为相对目录，则按照 `global` 中配置的 `deploy_dir` 生成
+- `data_dir`：指定数据目录。若不指定该字段或指定为相对目录，数据目录则按照 `global` 中配置的 `data_dir` 生成。该字段自集群版本 v4.0.14/v5.0.2 起生效。在早于该版本的集群中，该字段不生效。
```
Comment (TomShawn) [inline tiup/tiup-cluster-topology-reference.md:383]:
```suggestion
- `data_dir`：指定数据目录。若不指定该字段或指定为相对目录，数据目录则按照 `global` 中配置的 `data_dir` 生成。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**